lib/git_worktree.rb

```diff
@@ -139,7 +160,7 @@ def mv_dir(old_dir, new_dir)
 
   def fetch_and_merge
     fetch
-    commit = @repo.ref("refs/remotes/#{@remote_name}/master").target
+    commit = @repo.ref("refs/remotes/#{@remote_name}/#{@current_branch}").target
```

Comment:
No, a SHA is a perfectly reasonable ref, just as a branch or tag is.

I guess I don't understand what this is trying to solve based on your use cases in the PR body.